### PR TITLE
libturbojpeg: build it from sources.

### PR DESCRIPTION
**Pull request by gershonho**
Replaces pre-build libturbojpeg-binaries with libturbojpeg sources
accompanied with our custom CmakeLists.txt. Also drops the original
libjpeg to be 100% sure it is not used by the build.

### CODE REVIEW

**Comment by StatusReport** [comment]
@gershonho since the PR is huge, please split to 3 commits:

- Addition of libturbojpeg.
- Changes in the build files.
- Removal of libjpeg.

**Comment by gershonho** [comment]
Done.

File: 3rdparty/libturbojpeg/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,71 @@
+# ----------------------------------------------------------------------------
+#  CMake file for libjpeg. See root CMakeLists.txt
```
Comment (StatusReport) [inline 3rdparty/libturbojpeg/CMakeLists.txt]: libturbojpeg

File: 3rdparty/libturbojpeg/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+#
+# ----------------------------------------------------------------------------
+project(${JPEG_LIBRARY})
+
+ocv_include_directories(${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_BINARY_DIR})
+
+set(LIB_TURBOJPEG_SOURCES 
+  jaricom.c jcarith.c jdarith.c
+  jcapimin.c jcapistd.c jccoefct.c jccolor.c 
+	jcdctmgr.c jchuff.c jcinit.c jcmainct.c jcmarker.c jcmaster.c
```
Comment (StatusReport) [inline 3rdparty/libturbojpeg/CMakeLists.txt]: Consistent indentation. Also for the rest of file - some of the indentations are done using tabs and some using spaces.

File: 3rdparty/libturbojpeg/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+	endif()
+else()
+  set(LIB_TURBOJPEG_SOURCES ${LIB_TURBOJPEG_SOURCES} jsimd_none.c)
+endif()
+
+# ----------------------------------------------------------------------------------
+#         Define the library target:
+# ----------------------------------------------------------------------------------
+
+add_library(${JPEG_LIBRARY} STATIC ${LIB_TURBOJPEG_SOURCES} )
```
Comment (StatusReport) [inline 3rdparty/libturbojpeg/CMakeLists.txt]: `})`

File: 3rdparty/libturbojpeg/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+if(NOT BUILD_SHARED_LIBS)
+  ocv_install_target(${JPEG_LIBRARY} EXPORT OpenCVModules ARCHIVE DESTINATION ${OPENCV_3P_LIB_INSTALL_PATH} COMPONENT dev)
+endif()
+
+
+install(FILES jconfig.h jerror.h jmorecfg.h jpeglib.h turbojpeg.h DESTINATION "${OPENCV_INCLUDE_INSTALL_PATH}/opencv2/${JPEG_LIBRARY}/")
+
+set(OPENCV_MODULES_PUBLIC ${OPENCV_MODULES_PUBLIC} "${JPEG_LIBRARY}" CACHE INTERNAL "List of OpenCV modules marked for export")
+
```
Comment (StatusReport) [inline 3rdparty/libturbojpeg/CMakeLists.txt]: remove

File: 3rdparty/libturbojpeg/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+ocv_warnings_disable(CMAKE_C_FLAGS -Wunused-parameter) # clang
+ocv_warnings_disable(CMAKE_C_FLAGS /wd4013 /wd4244 /wd4267) # vs2005
+
+set_target_properties(${JPEG_LIBRARY}
+  PROPERTIES OUTPUT_NAME ${JPEG_LIBRARY}
+  DEBUG_POSTFIX "${OPENCV_DEBUG_POSTFIX}"
+  COMPILE_PDB_NAME ${JPEG_LIBRARY}
+  COMPILE_PDB_NAME_DEBUG "${JPEG_LIBRARY}${OPENCV_DEBUG_POSTFIX}"
+  ARCHIVE_OUTPUT_DIRECTORY ${3P_LIBRARY_OUTPUT_PATH}
+  )
```
Comment (StatusReport) [inline 3rdparty/libturbojpeg/CMakeLists.txt]: Indentation

**Review by StatusReport — CHANGES_REQUESTED** [review]
Fix and squash.

File: 3rdparty/libturbojpeg/CMakeLists.txt (hunk 1 above)
Comment (gershonho) [reply 3rdparty/libturbojpeg/CMakeLists.txt]: Done.

File: 3rdparty/libturbojpeg/CMakeLists.txt (hunk 2 above)
Comment (gershonho) [reply 3rdparty/libturbojpeg/CMakeLists.txt]: Done.

File: 3rdparty/libturbojpeg/CMakeLists.txt (hunk 3 above)
Comment (gershonho) [reply 3rdparty/libturbojpeg/CMakeLists.txt]: Done.

File: 3rdparty/libturbojpeg/CMakeLists.txt (hunk 5 above)
Comment (gershonho) [reply 3rdparty/libturbojpeg/CMakeLists.txt]: Done.

File: 3rdparty/libturbojpeg/CMakeLists.txt (hunk 4 above)
Comment (gershonho) [reply 3rdparty/libturbojpeg/CMakeLists.txt]: Done.

**Comment by gershonho** [comment]
Done.

**Comment by StatusReport** [comment]
LGTM, up to the linker warnings you talked about. Did you solve them?

**Comment by gershonho** [comment]
Yes - once I have cleaned the project and the Xcode caches these warnings have disappeared.

**Review by StatusReport — APPROVED** [review]

**Pull request merged**